weave/ops_arrow/util.py

```diff
@@ -45,6 +46,10 @@ def _eq_null_consumer_helper(
 
 
 def not_equal(lhs: pa.Array, rhs: typing.Union[pa.Array, pa.Scalar]) -> pa.Array:
+    # this unboxes the null scalar if it is boxed
+    if rhs == None:
```

Comment:
is this something we need to do for all lhs operands?

Reply:
No bc the lhs are already arrays